include/gdf/cffi/types.h

```diff
@@ -20,86 +24,136 @@ typedef enum {
-    GDF_HASH_IDENTITY,  /**< Identity hash function that simply returns the key to be hashed */
+    GDF_HASH_MURMUR3=0, ///< Murmur3 hash function; see @url https://en.wikipedia.org/wiki/MurmurHash#MurmurHash3
+    GDF_HASH_IDENTITY,  ///< Identity hash function that simply returns the key to be hashed
 } gdf_hash_func;
 
+/**
+ * @brief The resolution, or unit, for durations of time, used alongside duration values
+ */
 typedef enum {
-	TIME_UNIT_NONE=0, // default (undefined)
-	TIME_UNIT_s,   // second
-	TIME_UNIT_ms,  // millisecond
-	TIME_UNIT_us,  // microsecond
-	TIME_UNIT_ns   // nanosecond
+	TIME_UNIT_NONE=0, ///< time unit is undefined/unknown; this is the implicit default
+	TIME_UNIT_s,      ///< seconds
+	TIME_UNIT_ms,     ///< milliseconds (10^{-3} seconds)
+	TIME_UNIT_us,     ///< microseconds (10^{-6} seconds)
+	TIME_UNIT_ns      ///< nanoseconds  (10^{-9} seconds)
 } gdf_time_unit;
 
+/**
+ * @brief Potential Auxiliary information regarding a datum in a libgdf column.
+ *
+ * @note held either at the single-element or whole-column level.
+ *
+ */
 typedef struct {
 	gdf_time_unit time_unit;
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
-typedef struct gdf_column_{
-    void *data;                       /**< Pointer to the columns data */ 
-    gdf_valid_type *valid;            /**< Pointer to the columns validity bit mask where the 'i'th bit indicates if the 'i'th row is NULL */
-    gdf_size_type size;               /**< Number of data elements in the columns data buffer*/
-    gdf_dtype dtype;                  /**< The datatype of the column's data */
-    gdf_size_type null_count;         /**< The number of NULL values in the column's data */
-    gdf_dtype_extra_info dtype_info;
-    char *			col_name;			// host-side:	null terminated string
-} gdf_column;
-
-/* --------------------------------------------------------------------------*/
+/**
+ * @brief The fundamental, columnar format of data the GDF library works with.
+ *
+ * A gdf_column_ may originate in a RDBMS-like schema table; it may be the intermediary result
+ * within an execution plan; or it may be the result of non-DBMS related computation.
+ */
+ typedef struct gdf_column_{
+     void *data;
+         ///< Type-erased pointer to the column's raw data - which is a consecutive sequence
+         ///< with no gaps of elements of the type represented by `dtype`.
+         ///<
+         ///< @todo There are currently no formal alignment requirements, but it seems the
+         ///< implementation may implicitly be assuming alignment to the size of the relevant
+         ///< type.
+         ///< @todo Can this be NULL? What about after "construction"?
+         ///< @todo Is this always in device memory?
+
+     gdf_valid_type *valid;
+         ///< a pseudo-column of `size` bits, packed into bytes (with in-byte order from
```

Comment:
More detail about the bit/byte ordering of the valid mask would be useful for new developers. See the Apache Arrow description of this ordering here: https://arrow.apache.org/docs/memory_layout.html